packages/models/src/Action.ts

```diff
@@ -24,12 +24,20 @@ export enum ConditionType {
   GREATER_THAN_OR_EQUAL = "GREATER_THAN_OR_EQUAL",
   LESS_THAN = "LESS_THAN",
   LESS_THAN_OR_EQUAL = "LESS_THAN_OR_EQUAL",
+  STRING_EQUAL = "STRING_EQUAL",
+}
+
+export enum ComparisonType {
+  NUMBER_OF_ITEMS = "NUMBER_OF_ITEMS",
+  NUMERIC_VALUE = "NUMERIC_VALUE",
+  STRING = "STRING",
 }
 
 export interface Condition {
   entityId: string
```

Comment:
IIRC, the main reason(s) for using a stringValue field as opposed to a type field were:
- the usage of a type field was discussed in the past (before I even got here), and for whatever reason, was decided against. My assumption was that there was a concensus (or at least a compelling reason) not to use the type field, and was continuing in that pattern rather than revisiting past decisions.
- adding a type field would require a bunch of refactoring (not only here, but service side as well), that I wasn't really prepared to do at the time, since I was still getting up to speed on both this as well as service code.  Combining that refactoring effort along with these string matching changes wouldn't have been the best idea at the time.

If we do want to introduce a type field now that the condition logic is getting even richer than before, I certainly invite that discussion, but would say to do it separate from this PR.